common/unittest/blocks_allocator.cpp

```diff
@@ -0,0 +1,70 @@
+#include <gtest/gtest.h>
+#include <memory>
+
+#include "../blocks_allocator.h"
+
+namespace
+{
+
```

Comment:
These are good. Though I would add a test that covers a case where `number_first_reserved` gets equal to `blocks_total_`, smth like this should work:
```cpp
TEST(BlocksAllocator, SequentialAllocationFull)
{
	size_t blocks_number = 1024;
	uint32_t reserved = blocks_number;

	size_t size = common::allocator::BlocksAllocator<64>::GetBufferSize(blocks_number);
	std::unique_ptr<uint8_t[]> buffer = std::make_unique<uint8_t[]>(size);
	common::allocator::BlocksAllocator<64> allocator;
	allocator.Init(buffer.get(), size, reserved);

	// no more blocks shoudl be available
	ASSERT_EQ(allocator.Allocate(), common::allocator::BlocksAllocator<64>::null_block);
	ASSERT_EQ(allocator.Stat(), std::make_pair(blocks_number, blocks_number));

	for (size_t i = 0; i < reserved; i++)
	{
		ASSERT_EQ(allocator[i], buffer.get() + 64 * i);
	}
}
```